### PR TITLE
Add manual trigger targets for (support) docker image builds

### LINKED ISSUE #522: Generate fluxcd/website docker images through GH Action
As a follow-up to #502 - let's generate fluxcd/website docker images through a GH workflow. Discussed in https://cloud-native.slack.com/archives/C01A402P0R2/p1631017482043800

### PR DESCRIPTION
**Pull request by kingdonb**
Fixes #522

This PR will make the `docker-push-hugo` and `docker-push-support` targets triggerable as `workflow_dispatch` actions, through the Github Actions workflow tab on GitHub.

I tested this out on the `kingdon` branch of same fork of the website repo, the results are here: https://hub.docker.com/r/kingdonb/website/tags?page=1&ordering=last_updated

You can set `image: kingdonb/website:hugo-support` in `okteto.yml` and run `okteto up` to see this in action right now (the rebuilt image is what is required to make the `okteto up` or `make docker-serve` targets function properly)

### CODE REVIEW

**Comment by kingdonb** [comment]
For a more thorough test, fork and build it for yourself, with your own docker hub credentials in `DOCKER_FLUXCD_USER` and `DOCKER_FLUXCD_PASSWORD`

Then in that case you may need to reset this variable in the `Makefile` in your fork, and in `Dockerfile`, but I think I have narrowed it that far:

```Makefile
DEV_IMAGE_REGISTRY_NAME    ?= kingdonb
```

```Dockerfile
ARG HUGO_VERSION
FROM kingdonb/website:hugo-${HUGO_VERSION}-extended
```

I would have made this into a parameter but it serves no functional purpose to do so for the upstream, so I left it alone.

(Use your own Docker Hub registry name in the files above, instead of `kingdonb`)

When this is merged, you can simply go to the GitHub Actions tab on the FluxCD/website repo and find the "Push updated Docker hugo version" target, then run it manually. A few minutes later, when it hopefully succeeded, you can find the "Push updated hugo support image" target that depends on the first target, and run it, (they are implicitly dependent, so it is incumbent on maintainers to know about this and when each action should be run manually. Maybe later we can script it.)

Building Hugo takes a long time, longer than 5m, and it is something I surmise we won't need to do that often as the Hugo version doesn't have to change all that often, compared to the things that are in our `Dockerfile` that might need to change more often.

Both tasks take longer than a minute, both are ordered dependencies of our `make serve` action, and now you don't have to sit for both when you don't need both. I hope this can be merged straightforwardly.

Then, a maintainer with access to trigger `workflow_dispatch` targets should run the `Push updated Docker hugo version` target (once now, and again whenever HUGO_VERSION is updated in `netlify.toml`)

And also should run `Push updated hugo support image` whenever a Dockerfile, python, or npm dependency is updated.

I will update the docs to reflect the new GitHub Actions workflow for keeping docker-support updated in a separate PR, but it's going to read something fairly close to the tl;dr above.

I think you can run both actions after merging this @scottrigby and we will see updated images showing up in fluxcd/website ([here](https://hub.docker.com/r/fluxcd/website/tags?page=1&ordering=last_updated)) – then people can just run `okteto up` without knowing anything, and start making changes in there to test out updates to the website.

The output of my test runs are visible [here](https://hub.docker.com/r/kingdonb/website/tags?page=1&ordering=last_updated) as docker images that got pushed from the kingdonb branch, which is based on this one, but with a few places changed from `fluxcd` to `kingdonb`

**Comment by kingdonb** [comment]
The expected output of the test from top-post:

```
$ okteto up
⠹ Pulling images... W1001 23:33:38.249477   90600 transport.go:260] Unable to cancel request for *exec.roundTripper
W1001 23:33:38.257185   90600 transport.go:260] Unable to cancel request for *exec.roundTripper
⠸ Pulling images... W1001 23:34:08.258864   90600 transport.go:260] Unable to cancel request for *exec.roundTripper
 ✓  Images successfully pulled
 ✓  Files synchronized
    Context:   oidc@moo
    Namespace: default
    Name:      hugo
    Forward:   1313 -> 1313

make: Nothing to be done for 'theme'.
hack/adopters.py
hack/gen-content.py
hack/import-calendar.py
hack/import-flux2-assets.sh
############################################################################################################################################################## 100.0%
############################################################################################################################################################## 100.0%
# Git Repositories
############################################################################################################################################################## 100.0%
# Helm Repositories
############################################################################################################################################################## 100.0%
# Helm Charts
############################################################################################################################################################## 100.0%
# Object storage buckets
############################################################################################################################################################## 100.0%
############################################################################################################################################################## 100.0%
# Kustomization
############################################################################################################################################################## 100.0%
############################################################################################################################################################## 100.0%
# Helm Releases
############################################################################################################################################################## 100.0%
############################################################################################################################################################## 100.0%
# Event
############################################################################################################################################################## 100.0%
# Alert
############################################################################################################################################################## 100.0%
# Provider
############################################################################################################################################################## 100.0%
# Receiver
############################################################################################################################################################## 100.0%
############################################################################################################################################################## 100.0%
# Image Repositories
############################################################################################################################################################## 100.0%
# Image Policies
############################################################################################################################################################## 100.0%
############################################################################################################################################################## 100.0%
# Image Update Automations
hugo server \
	--bind 127.0.0.1 \
	--buildDrafts \
	--buildFuture \
	--disableFastRender
Start building sites …
hugo v0.88.1+extended linux/amd64 BuildDate=2021-10-01T20:58:11Z

                   | EN
-------------------+------
  Pages            | 307
  Paginator pages  |   1
  Non-page files   |   5
  Static files     | 124
  Processed images |   0
  Aliases          |   1
  Sitemaps         |   1
  Cleaned          |   0

Built in 2983 ms
WARN 2021/10/02 03:34:28 Unsupported symlink found in "/site/themes/docsy/assets/vendor/bootstrap-4.6.0", skipping.
WARN 2021/10/02 03:34:28 Unsupported symlink found in "/site/themes/docsy/assets/vendor/Font-Awesome-5.15.3", skipping.
Watching for changes in /site/{assets,content,i18n,layouts,package.json,static,themes}
Watching for config changes in /site/config.toml, /site/themes/docsy/config.toml
Environment: "development"
Serving pages from memory
Web Server is available at //localhost:1313/ (bind address 127.0.0.1)
Press Ctrl+C to stop
```

(You can then find the website at http://localhost:1313)

The same thing should work with `docker-serve` as long as you have updated the line in your `Makefile` as above, as in the `kingdon` branch:

```
DEV_IMAGE_REGISTRY_NAME    ?= kingdonb
```

* Note this is literally `kingdonb` as in `docker.io/kingdonb/website:hugo-support` unless you have built this image for yourself.

**Comment by kingdonb** [comment]
I just tested `make docker-serve` – it works, it takes a while on my machine, but it works!

`okteto up` might be faster because the remote machine that it runs on is itself a much faster machine than my laptop. This method should work with any Kubernetes cluster, with no prior setup required.

**Review by dholbach — APPROVED** [review]
Great work @kingdonb !

**Pull request merged**
